### PR TITLE
Testing APi unit tests

### PR DESCRIPTION
**Pull request by narekhovhannisyan**
## Motivation

Cover testing api with unit tests, fix minor changes.

## Changes

- Implemented tests for testing API
- Implemented tests for messages API
- Implemented tests for attachments API
- Implemented tests for inboxes API
- Implemented tests for projects API
- Added minor fixes to mailtrap client.
- Added test cases for mailtrap client.
- Added test cases for axios-logger.

### CODE REVIEW

File: src/__tests__/lib/api/resources/Inboxes.test.ts [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+  });
+
+  describe("resetCredentials(): ", () => {
+    it("succesfully resets credentials.", async () => {
+      const inboxId = 1;
+      const endpoint = `${GENERAL_ENDPOINT}/api/accounts/${accountId}/inboxes/${inboxId}/all_read`;
+
+      mock.onPatch(endpoint).reply(200, responseData);
+
+      const result = await inboxesAPI.markAsRead(inboxId);
```
Comment (mklocek) [inline src/__tests__/lib/api/resources/Inboxes.test.ts:292]: This should test `resetCredentials` but tests `markAsRead`

```suggestion
  describe("resetCredentials(): ", () => {
    it("succesfully resets credentials.", async () => {
      const inboxId = 1;
      const endpoint = `${GENERAL_ENDPOINT}/api/accounts/${accountId}/inboxes/${inboxId}/all_read`;

      mock.onPatch(endpoint).reply(200, responseData);

      const result = await inboxesAPI.markAsRead(inboxId);
```

File: src/__tests__/lib/axios-logger.test.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { AxiosError } from "axios";
+import axiosLogger from "../../lib/axios-logger";
+import MailtrapError from "../../lib/MailtrapError";
+
+describe("lib/axios-logger: ", () => {
+  describe("handleSendingError(): ", () => {
+    it("re-throws error if it's not axios error.", () => {
+      const unknownError = new Error("mock-error");
+
```
Comment (mklocek) [inline src/__tests__/lib/axios-logger.test.ts:9]: Missing `expect.assertions(1)`

**Review by mklocek — APPROVED** [review]

File: src/__tests__/lib/api/resources/Attachments.test.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      const expectedErrorMessage = "Request failed with status code 404";
+
+      expect.assertions(2);
+
+      try {
+        await attachmentsAPI.get(inboxId, messageId, attachmentId);
+      } catch (error) {
+        expect(error).toBeInstanceOf(MailtrapError);
+
+        if (error instanceof MailtrapError) {
```
Comment (VladimirTaytor) [inline src/__tests__/lib/api/resources/Attachments.test.ts:109]: #q Why do we need to always check if error is instance of MailtrapError 
- We do have assertion before it 🤔

**Review by VladimirTaytor — APPROVED** [review]

File: src/__tests__/lib/api/resources/Attachments.test.ts (hunk 3 above)
Comment (narekhovhannisyan) [reply src/__tests__/lib/api/resources/Attachments.test.ts:109]: @VladimirTaytor typescript requires to check error instance before accessing it

**Pull request merged**